Added the package WordingStatus

evandrocoan opened the pull request:
- [x] I'm the package's author and/or maintainer.
- [x] I have have read [the docs][1].
- [x] I have tagged a release with a [semver][2] version number.
- [x] My package repo has a description and a README describing what it's for and how to use it.
- [x] My package doesn't add context menu entries. *
- [x] My package doesn't add key bindings. **
- [x] Any commands are available via the command palette.
- [x] Preferences and keybindings (if any) are listed in the menu and the command palette, and open in split view.
- [x] If my package is a syntax it doesn't also add a color scheme. ***
- [x] I use [.gitattributes][3] to exclude files from the package: images, test files, sublime-project/workspace.

My package is a rewrite of WordCount (https://github.com/wbond/package_control_channel/issues/6907).

Fixes https://github.com/evandrocoan/WordingStatus/issues/5

[1]: https://packagecontrol.io/docs/submitting_a_package
[2]: https://semver.org
[3]: https://www.git-scm.com/docs/gitattributes#_export_ignore

braver commented on repository/w.json:
```diff
@@ -759,6 +759,18 @@
+		{
+			"name": "WordingStatus",
+			"details": "https://github.com/evandrocoan/WordingStatus",
+			"homepage": "https://github.com/evandrocoan/WordingStatus",
```
That "homepage" is not necessary since it's just the GitHub repo url.

braver commented:
Why should this exist next to the existing WordCount package? Can't your rewrite be contributed back to that repository?

braver commented:
Oh, that's a "Tito" package, and it's archived... of course you can't contribute back. Shall we just replace the existing package with yours?

evandrocoan commented:
Some may like the original. My code was a complete rewrite, and the functional requirements should be backward compatible, but there is a non-functional requirement not respected. For big files, my version does not often refresh, while the original version refreshes the exact word counting within the same period.

braver commented:
Honestly I'd like to phase out dead and archived packages and yours offers the exact same functionality (with minor changes that sounds like improvements to me). What you could do is ship as a major version upgrade, and have an upgrade-message explaining the take-over, and steps users can take to manually install the old package (e.g. check out that repo in the packages dir).

braver commented:
Maybe one place where your package can be improved is in factoring out the Preferences class. I don't know why it exists in the original package. Sublime's interface to settings serves the values straight from memory, and updates the values immediately when they change: you cannot improve upon the performance of the API. And the interface also intelligently supports defaults. This would significantly reduce your lines of code, reduce potential misdirection, and probably make things like changing settings, view-only settings, and project-level settings work automatically and more reliably than your own implementation here.

evandrocoan commented:
The CI test failed but it does not look related to may changes: 
![image](https://user-images.githubusercontent.com/5332158/206057429-639e7ee8-de90-4c62-98a9-5d134ce646d5.png)

braver commented:
Your changes don't look good right now, you're touching more than just your package. 

There are two possible approaches:

1. We keep the old name, but update the entry to point to your repo (and maybe some other adjustments). 
2. We remove the old entry and add yours, effectively changing the name (doesn't seem necessary to me). You would also need to add a "previous_names" in your entry with the name of the old entry in it, to transfer current installations to your package. 

Approach 1 seems simplest to me.

evandrocoan commented:
As I already renamed the package, I am keeping the new name, and I added the "previous_names" key.

I also rebased with `develop` to try to fix the python error, but now looks like another package is not correctly sorted:			
![image](https://user-images.githubusercontent.com/5332158/206337301-40c3ec0d-161d-4d52-8c39-89b861afa413.png)

FichteFoll commented:
Master was fixed, please rebase again.

braver commented:
Looks good, let's roll 👍🏻
